packages/flutter_tools/lib/src/android/android_studio.dart

```diff
@@ -107,11 +120,19 @@ class AndroidStudio implements Comparable<AndroidStudio> {
       final int major = version.major;
       final int minor = version.minor;
       if (platform.isMacOS) {
```

Comment:
ya, if you're going to do all this, we might as well extract the whole path defining into the constructors

Reply:
> ya, if you're going to do all this, we might as well extract the whole path defining into the constructors

Can you elaborate this?

Reply:
I meant it's a bit strange that the variable is named in such a generic way (pathSelectorPath) but is both nullable and only used on macs. It's also not very clear the distinction between the instance variables 'directory' and 'pathSelectorPath' from a first read. 

Since 'directory' is selected from fromMacOSBundle and fromHomeDot, perhaps just follow that pattern with _pluginsPath and just make that selected from the constructors too. Also feel free to rename them too to something like 'applicationPath' and 'pluginsPath' or something.